tests/std/tests/P0896R4_counted_iterator_death/test.cpp

```diff
@@ -0,0 +1,248 @@
+        return *_ptr;
+    };
+    simple_input_iter& operator++() {
+        ++_ptr;
+        return *this;
+    };
+    simple_input_iter operator++(int) {
+        simple_input_iter _tmp = *this;
+        ++_ptr;
+        return _tmp;
+    };
+
+    bool operator==(simple_input_iter const&) const = default;
+
+    int* _ptr = nullptr;
+};
+
+void test_case_construction_negative_length() {
+    counted_iterator<int*> cit{globalArray, -1}; // counted_iterator requires positive length n
+    (void) cit;
+}
+
+void test_case_operator_dereference_value_initialized_iterator() {
+    counted_iterator<int*> cit{}; // note: for IDL to work correctly, default-init and value-init are equivalent
+    (void) (*cit); // cannot dereference value-initialized counted_iterator
+}
+
+void test_case_operator_dereference_end_iterator() {
+    counted_iterator<int*> cit{globalArray, 0};
+    (void) (*cit); // cannot dereference end counted_iterator (per proposed resolution of LWG-3472)
+}
+
+void test_case_operator_preincrement_value_initialized_iterator() {
+    counted_iterator<int*> cit{};
+    ++cit; // cannot increment value-initialized counted_iterator
+}
+
+void test_case_operator_preincrement_value_initialized_input_iterator() {
+    counted_iterator<simple_input_iter> cit{};
+    ++cit; // cannot increment value-initialized counted_iterator
+}
+
+void test_case_operator_preincrement_after_end() {
+    counted_iterator<int*> cit{globalArray, 0};
+    ++cit; // cannot increment counted_iterator past end
+}
+
+void test_case_operator_preincrement_after_end_input_iterator() {
+    counted_iterator<simple_input_iter> cit{simple_input_iter{globalArray}, 0};
+    ++cit; // cannot increment counted_iterator past end
+}
+
+void test_case_operator_increment_value_initialized_iterator() {
+    counted_iterator<int*> cit{};
+    cit++; // cannot increment value-initialized counted_iterator
+}
+
+void test_case_operator_increment_value_initialized_input_iterator() {
+    counted_iterator<simple_input_iter> cit{};
+    cit++; // cannot increment value-initialized counted_iterator
+}
+
+void test_case_operator_increment_after_end() {
+    counted_iterator<int*> cit{globalArray, 0};
+    cit++; // cannot increment counted_iterator past end
+}
+
+void test_case_operator_increment_after_end_input_iterator() {
+    counted_iterator<simple_input_iter> cit{simple_input_iter{globalArray}, 0};
+    cit++; // cannot increment value-initialized counted_iterator
+}
+
+void test_case_operator_predecrement_before_begin() {
+    counted_iterator<vit> cit{begin(checkedArray), 5};
+    --cit; // cannot decrement counted_iterator before begin
+}
+
+void test_case_operator_decrement_before_begin() {
+    counted_iterator<vit> cit{begin(checkedArray), 5};
+    cit--; // cannot decrement counted_iterator before begin
+}
+
+void test_case_operator_advance_value_initialized_iterator() {
+    counted_iterator<int*> cit{};
+    cit += 1; // cannot seek value-initialized counted_iterator
+}

```

Comment:
We should have tests to verify that `+= 0` etc. are fine for value-initialized; see `tests\std\tests\VSO_0000000_string_view_idl\test.cpp` for examples. I think it would be fine to put those in the non-death test though.

Reply:
Added a test